Python/bytecodes.c

```diff
@@ -993,6 +994,25 @@ dummy_func(
             }
         }
 
+        inst(LOAD_LOCALS, ( -- locals)) {
```

Comment:
Is it possible to do a scan of the entire class/function first and add the type variables to the fastlocals array?

That way, we can reuse `LOAD_FAST` and not have to add a new opcode, which slows the interpreter down (albeit very slightly).

Reply:
It's not just about type variables. The new opcodes are intended solve this problem:

```
class X:
    y = 2
    type Alias = y
```
Because the RHS of `Alias` is lazily evaluated, we wrap it in a function-like scope, but normally function scopes can't see names defined in an enclosing class scope, so under a naive implementation, evaluating the value of `X.Alias` would fail. The two new opcodes and the new attribute on function objects provide a mechanism that allows these nested functions to see names in the enclosing class scope. (cc @larryhastings who had to solve a very similar problem for PEP-649, and who contributed part of the implementation here.)

We could possibly make LOAD_LOCALS an intrinsic if the new opcode is a big concern, though it would be more awkward to implement.

Reply:
I think it's fine. One opcode shouldn't hurt perf by much (if at all measurable).

Reply:
Good point. An intrinsic would be nice, but I'm not sure that's possible, since this code needs access to the frame, which I don't think is available inside an intrinsic. Alternatively, I could add a separate opcode, but it feels like a waste of an opcode for such an edge case.

I think I can instead adjust the compiler so that LOAD_LOCALS can simply load the locals only.

Reply:
Just pushed that.